docs/simple_switch.md

```diff
@@ -656,3 +656,31 @@ if `max >= 1`, otherwise `base`.
 
 The type `O` of the `result`, `T` of `base`, and `M` of `max` are
 restricted to be `bit<W>` for `W <= 64`.
+
+
+### BMv2 `direct_counter` implementation notes
+
+If a table `t` has a `direct_counter` object `c` associated with it by
+having a table property `counters = c` in its definition, the BMv2
+v1model implementation behaves as if every action for that table
+contains exactly one call to `c.count()`, whether it has none, one, or
+more than one.
+
+
+### BMv2 `direct_meter` implementation notes
+
+If a table `t` has a `direct_meter` object `m` associated with it by
+having a table property `meters = m` in its definition, then _at least
```

Comment:
I was not aware of this, what does the compiler do if there is no call to `m.read(...)`? Does it complain because it cannot deduce the output field?

Reply:
I believe the latest p4c issues a CompilerBug and asserts.  I can create a separate issue on p4c for that, since it shouldn't be a CompilerBug, but ideally a slightly more informative message explaining why the program will not compile.

Reply:
Yes, CompilerBug today.  Issue here: https://github.com/p4lang/p4c/issues/1829